emlib: warnings when using GCC 7

#### Description
This error occurs when building the default example for the ikea-tradfri board when using GCC 7.2.0:


#### Steps to reproduce the issue

Use GCC 7.2.0
`make BOARD=ikea-tradfri -C examples/default`

#### Expected results

No compilation errors

#### Actual results

```
/home/jgn/work/src/riot/examples/default/bin/pkg/ikea-tradfri/emlib/emlib/src/em_cmu.c: In function ‘CMU_ClockSelectSet’:
/home/jgn/work/src/riot/examples/default/bin/pkg/ikea-tradfri/emlib/emlib/src/em_cmu.c:3463:14: error: this statement may fall through [-Werror=implicit-fallthrough=]
       selReg = (selReg == NULL) ? &CMU->LFACLKSEL : selReg;
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jgn/work/src/riot/examples/default/bin/pkg/ikea-tradfri/emlib/emlib/src/em_cmu.c:3480:5: note: here
     case CMU_LFECLKSEL_REG:
     ^~~~
/home/jgn/work/src/riot/examples/default/bin/pkg/ikea-tradfri/emlib/emlib/src/em_cmu.c:3481:14: error: this statement may fall through [-Werror=implicit-fallthrough=]
       selReg = (selReg == NULL) ? &CMU->LFECLKSEL : selReg;
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jgn/work/src/riot/examples/default/bin/pkg/ikea-tradfri/emlib/emlib/src/em_cmu.c:3488:5: note: here
     case CMU_LFBCLKSEL_REG:
     ^~~~
cc1: all warnings being treated as errors
```

#### Versions

arm-none-eabi-gcc (Gentoo 7.2.0 p1.1) 7.2.0

#### Issue references

#8265

--- a/pkg/gecko_sdk/Makefile
+++ b/pkg/gecko_sdk/Makefile
@@ -1,6 +1,6 @@
 PKG_NAME=gecko_sdk
 PKG_URL=https://github.com/basilfx/RIOT-gecko-sdk
-PKG_VERSION=f8a90e4db4680984b5f9d3dd44ca528772458456
+PKG_VERSION=d6a90efb97f175101821304e6e5a1ba20eef02aa
 PKG_LICENSE=Zlib
 
 ifneq ($(CPU),efm32)

